### ISSUE TITLE
Promise should be rejected if connection was refused

### ISSUE THREAD

**Issue by marius-klimantavicius** [pyrsmk/qwest#49]
Browser: Google Chrome 40. 
If a request is made to a resource that is not accessible (i.e. server is not running) then 'net::ERR_CONNECTION_REFUSED' is returned but the promise is not resolved/rejected. Expected the promise to be rejected.

**Comment by pyrsmk** [comment]
The `catch` promise is not called? Weird...
Can you put a test case online?

**Comment by marius-klimantavicius** [comment]
In my case I had set attempts to null, with huge timeout and that seemed like a single failed request without rejecting a promise. Setting attempts to 1 fixed my issue.

**Comment by pyrsmk** [comment]
I should investigate that. I reopen it to have a reminder. Thanks for the explanation ;)

**Comment by marius-klimantavicius** [comment]
Yeah it seems that some kind of an issue is still present - the promise fails only after timeout and not after getting connection error.
I have a test case:
http://jsbin.com/fazojefoga/1/edit?html,js,console

When timeout is relatively long (in my dev env for debugging purposes I set it to 10 min) running the code gives us error in console almost immediately, however for actual promise error we have to wait till timeout expires.

**Comment by pyrsmk** [comment]
In fact... It's weird that the throwed error shows immediatly in the console with a timeout of 10min. It seems to be another bug. Because, by a matter of design, it's normal that the error promise is throwed after the timeout. It's coded like that. Anyway, I'll plan that for future `1.6` release ;)

**Comment by willdady** [comment]
Not sure if it helps or not but I just encountered this. I have my app configured to use CORS and I just so happened to forget to start my backend server and got 'net::ERR_CONNECTION_REFUSED' on the preflight OPTIONS request. Qwest didn't catch the error.

**Comment by pyrsmk** [comment]
Thanks for the report ;)

In fact, we won't fix the bug for the current major version. We are working on a complete rewrite and we will implement promises with ES6 directly (or an external library). So all problems related to promises will be fixed at once ^^

**Comment by pyrsmk** [comment]
A future release coming. This issue should be fixed by now, please reopen if you encounter this bug with qwest 2.
